src/scheduler/server_info.c

```diff
@@ -3912,6 +3918,41 @@ create_resource_assn_for_node(node_info *ninfo)
 		}
 	}
 
+	/* Lastly if restrict_res_to_release_on_suspend is set, suspended jobs may not have released all their resources
+	 * This is tricky since a suspended job knows what resources they released.
+	 * We need to know what they didn't release to account for in the nodes resources_assigned
+	 * Also, we only need to deal with suspended jobs outside of reservations since resources for reservations were handled above.
+	 */
+	if (ninfo->num_susp_jobs > 0) {
```

Comment:
we should probably only go inside this block if rrtros is set

Reply:
Since we discussed offline that we cannot do this as someone might turn off rrtros while there are jobs suspended with some resources retained, I take my comment back.